packages/components/tokens/components/button.js

```diff
@@ -1,72 +1,50 @@
 const elements = require('@compassion-gds/elements');
-const color = require('color');
+
+const FUNCTIONAL_COLOR = {
+  PRIMARY: '#0A4FBB',
+  SECONDARY: '#F3F6FB',
+  DISABLED: '#717E98',
+};
```

Comment:
Sorry but having these defined here does not make sense to me and  would be causing maintenance overhead

I think we should manage all hard values in a single place, if that place is `elements` then the "functional colors" should be moved there
We can still exclude them from the palette interface

I actually like the previous solution where values were computed using `lighten/darken` as that exactly the use cause for these state representing colors.
Also this computation happens only when tokens are bundled so it's no overhead to the end user, right?

Instead of trying to hard code things, we should try to support a solution that derives state related colors, based on the theme colors.
There should be a tool that helps identify what transformation a theme color needs to match the value in Figma

Reply:
cc @kendrick

Reply:
Sorry I missed the tag! I think I agree that `elements` is the right place for this, namespaced within a `functional` object to draw a line between brand colors and non-brand colors.

Reply:
So you mean:
`functional: {
    primary: '#0A4FBB',
    secondary: '#F3F6FB',
    disabled: '#717E98',
  }` in `elements/color`?

Reply:
I think one of the first two options below make the most sense (admittedly different from my comment above, represented by the 3rd option 😜):
- `elements.color.primary.plus1`
- `elements.color.primary.functional.plus1`
- ~`elements.color.functional.primary.plus1`~

… I don’t have much of a preference between those two. I’m definitely open to other arrangements if something makes more sense!

Reply:
Well, I think I had suggested something similar (`primaryPlus1`,` secondaryPlus2`, `disabled`), but in your cases, we also need to add `elements.color.primary.main` f.x.:
`primary: { main: '#005EB8', plus1: '#0A4FBB',  }`.
  The thing is we already have defined some `secondary/success/gray/etc.` colors like `secondary: '#0086BF',  secondaryMinus1: '#20556B',  secondaryPlus1: '#AFD8FF', success`.

Reply:
If we're going with hard values then we should decide:
1. Are we going with a nested structure like `primary.main`, `primary.plus1`? Or a preserve the current flat structure e.g. `secondaryMinus1`?
2. Should we use the plus/minus format? It seems a value like `secondaryPlus1` is easy to associate with Figma, but as a user of the library a name like `secondaryHover` or `secondaryActive` makes more sense to me 

Keeping a list of "functional" colors but naming them plus/minus does not really reveal their function, only that they shouldn't be used as brand colors

So I propose something in this direction: 
- Have `colors` that would represent brand colors: primary, secondary, success, error, info, black, gray ....
- Have `accents` that would represent plus/minus values: successMinus1, errorMinus1 etc...
  - or some other name like "shades" just not "functional" 

Then in `tokens` 
1. use a brand `colors` like we currently do 
2. use `accents` for state representing values like `hover: elements.accents.primaryPlus1` 

---

I like Kerim's idea as well, as it overall the same thing but with a nested structure 
- regular color would be `color: elements.color.primary.main`
- stateful color:  `hover: elements.color.primary.plus1`

Reply:
> Keeping a list of “functional” colors but naming them plus/minus does not really reveal their function
Love the way you’ve expressed this and totally agree.

Given the thoughts above, I prefer keeping both element colors and shades/accents together under the same ancestor, and then relying on tokens to establish the functional link (as in the hover example above). My only tweak is that we use `base` instead of `main`, only to follow convention elsewhere in the project.